components/compliance-service/inspec-agent/runner/runner.go

```diff
@@ -0,0 +1,661 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	neturl "net/url"
+	"os"
+	"strings"
+	"time"
+
+	uuid "github.com/gofrs/uuid"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/chef/automate/components/compliance-service/api/jobs"
+	"github.com/chef/automate/components/compliance-service/ingest/events/compliance"
+	"github.com/chef/automate/components/compliance-service/ingest/ingest"
+	"github.com/chef/automate/components/compliance-service/ingest/ingestic/mappings"
+	"github.com/chef/automate/components/compliance-service/inspec"
+	"github.com/chef/automate/components/compliance-service/inspec-agent/remote"
+	"github.com/chef/automate/components/compliance-service/inspec-agent/resolver"
+	"github.com/chef/automate/components/compliance-service/inspec-agent/types"
+	"github.com/chef/automate/components/compliance-service/scanner"
+	"github.com/chef/automate/lib/workflow"
+)
+
+var ListenPort int = 2133
+
+func InitWorkflowManager(w *workflow.WorkflowManager, workerCount int, ingestClient ingest.ComplianceIngesterClient,
+	scanner *scanner.Scanner, resolver *resolver.Resolver, remoteInspecVersion string) {
+	w.RegisterWorkflowExecutor("scan-job-workflow", &ScanJobWorkflow{})
+	w.RegisterTaskExecutor("create-child", &CreateChildTask{
+		scanner,
+	}, workflow.TaskExecutorOpts{Workers: 1})
+	w.RegisterTaskExecutor("resolve-job", &ResolveTask{
+		remoteInspecVersion,
+		scanner,
+		resolver,
+	}, workflow.TaskExecutorOpts{Workers: 1})
+	w.RegisterTaskExecutor("scan-job", &InspecJobTask{
+		ingestClient,
+		scanner,
+	}, workflow.TaskExecutorOpts{Workers: workerCount})
+	w.RegisterTaskExecutor("scan-job-summary", &InspecJobSummaryTask{
+		scanner,
+	}, workflow.TaskExecutorOpts{Workers: 1})
+}
+
+type ScanJobWorkflow struct{}
+type ScanJobWorkflowPayload struct {
+	OutstandingJobs  int
+	ParentJobID      string
+	ChildJobID       string
+	OverallJobStatus string
+}
+
+func (p *ScanJobWorkflow) OnStart(w workflow.WorkflowInstance,
+	ev workflow.StartEvent) workflow.Decision {
+
+	var job jobs.Job
+	err := w.GetParameters(&job)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to unmarshal job!")
+		return w.Complete()
+	}
+
+	if job.Recurrence != "" {
+		w.EnqueueTask("create-child", job)
+	} else {
+		w.EnqueueTask("resolve-job", job)
+	}
+
+	return w.Continue(&ScanJobWorkflowPayload{
+		0,
+		job.Id,
+		"",
+		types.StatusRunning,
+	})
+}
+
+func (p *ScanJobWorkflow) OnTaskComplete(w workflow.WorkflowInstance,
+	ev workflow.TaskCompleteEvent) workflow.Decision {
+
+	var payload ScanJobWorkflowPayload
+
+	if err := w.GetPayload(&payload); err != nil {
+		logrus.WithError(err).Fatal("Could not decode payload")
```

Comment:
⚠️ We don't want to use fatal here.  

I think we want to extend the workflow library to be able to do `return w.Fail(err)`